src/odemis/gui/model/__init__.py

```diff
@@ -307,6 +307,22 @@ def __init__(self, microscope):
             if self.spectrometer is None and self.spectrometers:
                 self.spectrometer = self.spectrometers[0]
 
+            # Check for the most known microscope types that the basics are there
+            required_roles = []
+            if self.role in ("secom", "delphi", "cryo-secom"):
+                required_roles += ["e-beam", "light", "stage", "focus"]
+                if self.role == "secom":
+                    required_roles += ["align", "se-detector"]
+            elif self.role in ("sparc", "sparc2"):
+                required_roles += ["e-beam", "mirror", "lens"]
+            elif self.role == "mbsem":
+                required_roles += ["e-beam", "stage"]
```

Comment:
In the latest yaml file the mbsem does not have a component with role "e-beam"

Reply:
The fastem-sim has it. I think that the normal one should also have it, so that the overview works. Actually, for now, the restrictions are very light for the FASTEM, but once the simulator is a little bit more fluffy, which should be able to add extra needed components.